MainProject/Factory.cs

```diff
@@ -0,0 +1,101 @@
+﻿using ClassLibrary;
+
+class Program
+{
+    static void Main(string[] args)
+    {
+        Factory iTechArt = new Factory
+        (
+            "iTechArt",
+            "Krakow",
+            1,
+            new Employee[]
+            {
+                new Employee ("John", "Doe", 23),
+                new Employee ("Helena", "Bug", 20),
+                new Employee ("Harry", "Potter", 19),
+                new Employee ("Hanna", "Smith", 27, 5, "Middle"),
+                new Employee ("Clark", "Kent", 30, 10, "Senior" ),
+                new Employee ("Elon", "Musk", 22, 1, "Junior"),
+                new Employee ( "Barak", "Obama", 35, 15, "Senior"),
+            }
+        );
+
+        Factory Tesla = new Factory
+        (
+            "Tesla",
+            "Warsaw",
+            3,
+            new Employee[]
+            {
+                new Employee ("Donald", "Tusk", 20),
+                new Employee ("Kate", "Smith", 18),
+                new Employee ("July", "Trump", 2, 5, "Middle"),
+                new Employee ("James", "Jobs", 30, 4, "Middle"),
+                new Employee ("Alex", "Taylor", 21, 1, "Junior"),
+            }
+        );
+
+        Factory Microsoft = new Factory
+        (
+            "Microsoft",
+            "Gdansk"
+        );
+
+        Console.WriteLine($"The number of employees: {iTechArt.ReturnNumberEmployees()}");
+        iTechArt.AddEmployee(new Employee("Drako", "Malfoy", 19));
+        iTechArt.ListEmployees();
+        iTechArt.ListEmployeesByPosition();
+    }
+}
+public class Factory
+{
+    private string factoryName;
+    private string city;
+    private int officeNumber = 1;
+    private Employee[] employees;
```

Comment:
Additional task. на 5 с плюсом, так сказать) 
1. Use List instead of array here. 
2. Refactor dedicated methods according to new type
3. Add method which removing employee from the list